pkg/service/auth.go

```diff
@@ -105,10 +107,12 @@ func (a *AuthService) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 	}
 
 	if payload, err = ioutil.ReadAll(http.MaxBytesReader(resp, req.Body, a.MaxHttpRequestBodySize)); err != nil {
-		switch fmt.Sprint(err) {
-		case "http: request body too large":
+		switch err.Error() {
+		case ERROR_MESSAGE_413:
+			logger.V(1).Info(ERROR_MESSAGE_413)
 			closeWithStatus(envoy_type.StatusCode_PayloadTooLarge, resp, ctx, nil)
 		default:
+			logger.V(1).Info(ERROR_MESSAGE_404)
```

Comment:
Maybe this one before lines 90 and 97? Then a third type `ERROR_MESSAGE_400` here?